plugin/pkg/scheduler/algorithmprovider/affinity/affinity.go

```diff
@@ -37,9 +37,9 @@ func affinityPredicates() util.StringSet {
 		"PodFitsResources",
 		"NoDiskConflict",
 		// Ensures that all pods within the same service are hosted on minions within the same region as defined by the "region" label
-		factory.RegisterFitPredicate("ServiceAffinity", algorithm.NewServiceAffinityPredicate(factory.PodLister, factory.ServiceLister, factory.MinionLister, []string{"region"})),
+		factory.RegisterFitPredicate("RegionAffinity", algorithm.NewServiceAffinityPredicate(factory.PodLister, factory.ServiceLister, factory.MinionLister, []string{"region"})),
```

Comment:
I'm supportive of spreading by arbitrary labels. A common use case on bare metal is spreading across racks (power and network domain) and bus bars (power domain), for example. 

I would just prefer not to use region and zone as examples, nor as defaults.

Another possibility is that we could support a few generic topology levels (level1, level2, level3, ..., level6) by default.